card/mod/standard/set/all/rich_html/form.rb

```diff
@@ -54,7 +54,14 @@ def form_for_multi
 
   def subcard_input_names
     return "" if !form_root_format || form_root_format == self
-    "#{@parent.subcard_input_names}[subcards][#{card.contextual_name}]"
+    "#{@parent.subcard_input_names}[subcards][#{name_in_form}]"
+  end
+
+  # If you use subfield cards to render a form for a new card
+  # then the subfield cards should be created on the new card not the existing
+  # card that build the form
+  def name_in_form
+    @native_mode == :new ? card.relative_name : card.contextual_name
```

Comment:
Yes, I did consider that. I also considered to punch myself in the groin. But after a second of reflection I rejected both ideas.  I don't know why exactly but I have a distant memory that I have done things like these before and their outcome was equally painful.